src/shared/label-util.c

```diff
@@ -115,3 +116,29 @@ int btrfs_subvol_make_label(const char *path) {
 
         return mac_smack_fix(path, 0);
 }
+
+static int label_pre(int dir_fd, const char *path, mode_t mode) {
+        return mac_selinux_create_file_prepare_at(dir_fd, path, mode);
+}
+
+static int label_post(int dir_fd, const char *path) {
+        mac_selinux_create_file_clear();
+        return mac_smack_fix_full(dir_fd, path, NULL, 0);
```

Comment:
hmm? my thinking was that selinux and smack would implement separate LabelOps vtables, and we'd hence do *either* selinux labelling *or* smack labelling, but never both.

Reply:
These functions are a noop if the relevant MAC is not enabled.

Reply:
but this is still weird: if we compile without any mac, then we'll still end up with a vtable, and that's something we shouldn't have i am sure.

Reply:
Why not? It'll just be a noop

Reply:
I added a shortcut to mac_init() to not register label_ops if no MAC is enabled.

Reply:
I'm not sure if these can be enabled/disabled/switched at runtime though? If that's the case then we'd always want to register label ops